File: python/mirage/core/nextcloud/exists.py
from mirage.accessor.nextcloud import NextcloudAccessor
from mirage.cache.index import NULL_INDEX
from mirage.core.nextcloud.stat import stat
from mirage.types import PathSpec


async def exists(accessor: NextcloudAccessor, path: PathSpec) -> bool:
    try:
        await stat(accessor, path, index=NULL_INDEX)
        return True
    except (FileNotFoundError, ValueError):
        return False

File: typescript/packages/node/src/core/nextcloud/exists.ts
import type { PathSpec } from '@struktoai/mirage-core'
import type { NextcloudAccessor } from '../../accessor/nextcloud.ts'
import { stat } from './stat.ts'

export async function exists(accessor: NextcloudAccessor, path: PathSpec): Promise<boolean> {
  try {
    await stat(accessor, path)
    return true
  } catch (error) {
    if ((error as { code?: string } | null)?.code === 'ENOENT') return false
    throw error
  }
}
